services/kvstoremanager/redis.go

```diff
@@ -150,3 +168,60 @@ func (m *redisManagerT) HSet(hash, key string, value interface{}) (err error) {
 	}
 	return err
 }
+
+func (m *RedisManager) ExtractJSONSetArgs(jsonData json.RawMessage) ([]string, error) {
+	key := gjson.GetBytes(jsonData, "message.key").String()
+	path := gjson.GetBytes(jsonData, "message.path").String()
+	jsonVal := gjson.GetBytes(jsonData, "message.value")
+
+	actualPath := "$" // root insert
+	if path != "" {
+		actualPath = fmt.Sprintf("$.%s", path)
+	}
+	args := []string{key, actualPath, jsonVal.String()}
+
+	if actualPath != "$" {
```

Comment:
can you add some comments here

Reply:
Added them. Can you check and let me know if they are ok ?